pkgs/development/libraries/tiledb/default.nix

```
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, zlib
+, lz4
+, bzip2
+, zstd
+, spdlog_0
+, tbb
+, openssl
+, boost
+, libpqxx
+, clang-tools
+, catch2
+, python
+, gtest
+}:
+
+stdenv.mkDerivation rec {
+  name = "tiledb";
+  version = "1.6.3";
+
+  src = fetchFromGitHub {
+    owner = "TileDB-Inc";
+    repo = "TileDB";
+    rev = version;
+    sha256 = "sha256:0dgvz34v1w1sd9zrxpf0is6m4k4b0nzllmwyrdbwsiclkaxqa0ha";
+  };
+
+  nativeBuildInputs = [ clang-tools cmake gtest ];
+
+  buildInputs = [ catch2 zlib lz4 bzip2 zstd spdlog_0 tbb openssl boost libpqxx python ];
+
+  # emulate the process of pulling catch down
+  postPatch = ''
+    mkdir -p build/externals/src/ep_catch
+    ln -sf ${catch2}/include/catch2 build/externals/src/ep_catch/single_include
+  '';
+
+  doCheck = false;
+
+  installTargets = "install-tiledb";
+
+  meta = with lib; {
+    description = "TileDB allows you to manage the massive dense and sparse multi-dimensional array data";
+    homepage = https://github.com/TileDB-Inc/TileDB;
+    license = licenses.mit;
```

```suggestion
    license = licenses.mit;
    platforms = [ "x86_64-linux"];
```
to avoid hydra trying to constantly attempt to build aarch, as it errors:
```
/build/source/external/src/blosc/shuffle.cc:280:4: error: #warning Hardware-acceleration detection not implemented for the target architecture. Only the generic shuffle/unshuffle routines will be available. [-Werror=cpp]
   #warning Hardware-acceleration detection not implemented for the target architecture. Only the generic shuffle/unshuffle routines will be available.
    ^~~~~~~
[ 96%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/__/external/src/blosc/shuffle-generic.cc.o
/build/source/external/src/blosc/shuffle.cc: In function 'blosc::shuffle_implementation_t blosc::get_shuffle_implementation()':
/build/source/external/src/blosc/shuffle.cc:290:22: error: unused variable 'cpu_features' [-Werror=unused-variable]
   blosc_cpu_features cpu_features = blosc_get_cpu_features();
                      ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

---

this should probably be upstreamed to the tiledb repo that their arm build has errors with gcc7

---

Ok.